File: qqbot-runtime/src/main/java/com/mieai/qqbot/runtime/security/AppSecretCipher.java
package com.mieai.qqbot.runtime.security;

import com.mieai.qqbot.persistence.bot.SecretCiphertext;

/** Encryption boundary for AppSecret persistence. */
public interface AppSecretCipher {
    SecretCiphertext encrypt(AppSecret secret, AppSecretBinding binding);

    AppSecret decrypt(SecretCiphertext ciphertext, AppSecretBinding binding);
}

File: qqbot-runtime/src/main/kotlin/com/mieai/qqbot/runtime/security/AppSecretCipher.kt
package com.mieai.qqbot.runtime.security
import com.mieai.qqbot.persistence.bot.SecretCiphertext
interface AppSecretCipher { fun encrypt(secret: AppSecret, binding: AppSecretBinding): SecretCiphertext; fun decrypt(ciphertext: SecretCiphertext, binding: AppSecretBinding): AppSecret }
